spec/ar_lazy_preload/active_record/relation_spec.rb

```diff
@@ -19,6 +19,12 @@
     it "raises exception on empty arguments" do
       expect { User.lazy_preload }.to raise_exception(ArgumentError)
     end
+
+    it "can lazy preload associations on association collection" do
```

Comment:
I think we could move this test to the separate file `spec/ar_lazy_preload/active_record/collection_proxy_spec.rb`.

Please make sure user` is created in the `let` statement - in the current implementation the test will behave differently when it's run as a part of the suite and separately (because in the second case there are no users in the DB).

Also, it would be great to check if preloading is still working here (context is initialized properly): I think the easiest way is to check if context was set up properly